### PR TITLE
fix: duplexConn ReadFrom and connect handler initial data coalescing

### PR DESCRIPTION
**Pull request by fortuna**
I'm also fixing the connect handler, which was not properly coalescing the initial data.

It was that fix that made me realize the issue with the duplexConn.
And I removed a test that was using an external service.

### CODE REVIEW

File: transport/stream.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (dc *duplexConnAdaptor) Write(b []byte) (int, error) {
 }
 func (dc *duplexConnAdaptor) ReadFrom(r io.Reader) (int64, error) {
 	// Make sure we prefer ReadFrom. Otherwise io.Copy will try WriteTo first.
-	if rf, ok := r.(io.ReaderFrom); ok {
+	if rf, ok := dc.w.(io.ReaderFrom); ok {
```
Comment (jyyi1) [inline transport/stream.go:56]: Can we add some tests for `duplexConnAdaptor`, to ensure that we won't introduce any regressions in the future.

File: x/httpproxy/connect_handler.go [hunk 2]
```diff
@@ -119,7 +119,16 @@ func (h *connectHandler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http
 
 	// Relay data between client and target in both directions.
 	go func() {
-		io.Copy(targetConn, clientRW)
+		// io.Copy prefers WriteTo, which clientRW implements. However,
+		// bufio.ReadWriter.ReadFrom issues an empty Write() call, which flushes
```
Comment (jyyi1) [inline x/httpproxy/connect_handler.go:123]: Do you mean `bufio.ReadWriter.WriteTo` here?

**Review by jyyi1 — COMMENTED** [review]
Thanks. I think we should add some tests, so we can catch it through some unit tests in the future.

File: x/httpproxy/connect_handler.go (hunk 2 above)
Comment (fortuna) [reply x/httpproxy/connect_handler.go]: Yes. Fixed.

File: transport/stream.go (hunk 1 above)
Comment (fortuna) [reply transport/stream.go:56]: done

**Review by jyyi1 — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**